onadata/apps/api/viewsets/xform_viewset.py

```diff
@@ -474,12 +474,17 @@ def retrieve(self, request, *args, **kwargs):
     def share(self, request, *args, **kwargs):
         self.object = self.get_object()
 
-        data = {}
-        for key, val in request.data.iteritems():
-            data[key] = val
-        data.update({'xform': self.object.pk})
+        usernames_str = request.data.get("users", request.data.get("username"))
+        if not usernames_str:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
 
-        serializer = ShareXFormSerializer(data=data)
+        data_list = []
+        for username in usernames_str.split(","):
+            data_list.append({"xform":    self.object.pk,
+                              "username": username,
+                              "role":     request.data.get("role")})
```

Comment:
this only needs to be fetched once, even though it's O(1) no the fetch cost is O(n), n=number of users